server/src/main/java/org/opensearch/index/compositeindex/startree/StarTreeIndexSettings.java

```diff
@@ -0,0 +1,101 @@
+import org.opensearch.common.settings.Setting;
+import org.opensearch.index.compositeindex.MetricType;
+import org.opensearch.search.aggregations.bucket.histogram.DateHistogramAggregationBuilder;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Index settings for star tree fields
+ *
+ * @opensearch.experimental
+ */
+public class StarTreeIndexSettings {
+    /**
+     * This setting determines the max number of star tree fields that can be part of composite index mapping. For each
+     * star tree field, we will generate associated star tree index.
+     */
+    public static final Setting<Integer> STAR_TREE_MAX_FIELDS_SETTING = Setting.intSetting(
+        "index.composite.star_tree.max_fields",
+        1,
+        1,
+        1,
+        Setting.Property.IndexScope,
+        Setting.Property.Final
+    );
+
+    /**
+     * This setting determines the max number of dimensions that can be part of star tree index field. Number of
+     * dimensions and associated cardinality has direct effect of star tree index size and query performance.
+     */
+    public static final Setting<Integer> STAR_TREE_MAX_DIMENSIONS_SETTING = Setting.intSetting(
+        "index.composite.star_tree.field.max_dimensions",
+        10,
+        2,
+        10,
+        Setting.Property.IndexScope,
+        Setting.Property.Final
+    );
+
+    /**
+     * This setting configures the default "maxLeafDocs" setting of star tree. This affects both query performance and
+     * star tree index size. Lesser the leaves, better the query latency but higher storage size and vice versa
+     * <p>
+     * We can remove this later or change it to an enum based constant setting.
+     *
+     * @opensearch.experimental
+     */
+    public static final Setting<Integer> STAR_TREE_DEFAULT_MAX_LEAF_DOCS = Setting.intSetting(
+        "index.composite.star_tree.default.max_leaf_docs",
+        10000,
+        1,
+        Setting.Property.IndexScope,
+        Setting.Property.Final
+    );
+
+    /**
+     * Default intervals for date dimension as part of star tree fields
+     */
+    public static final Setting<List<Rounding.DateTimeUnit>> DEFAULT_DATE_INTERVALS = Setting.listSetting(
+        "index.composite.star_tree.field.default.date_intervals",
+        Arrays.asList(Rounding.DateTimeUnit.MINUTES_OF_HOUR.shortName(), Rounding.DateTimeUnit.HOUR_OF_DAY.shortName()),
+        StarTreeIndexSettings::getTimeUnit,
+        Setting.Property.IndexScope,
+        Setting.Property.Final
+    );
+
+    /**
+     * Default metrics for metrics as part of star tree fields
+     */
+    public static final Setting<List<MetricType>> DEFAULT_METRICS_LIST = Setting.listSetting(
+        "index.composite.star_tree.field.default.metrics",
+        Arrays.asList(
+            MetricType.AVG.toString(),
+            MetricType.COUNT.toString(),
+            MetricType.SUM.toString(),
+            MetricType.MAX.toString(),
+            MetricType.MIN.toString()
```

Comment:
The defaults should ensure that we provide a good set of features without adding a lot of overhead. I'm not very sure on the kind of overhead each MetricType adds to storage, and computation in indexing time, but if we can keep the defaults to have fair trade-off between fast query, and storage/compute cost, that would be ideal.

I'm probably fine with this list, but would like to hear on the difference in overhead it creates if one is added/removed from the defaults list.

Reply:
The overhead is linear for metrics, as an additional `DocValues` field will be created and the size rather depends on the number of star tree documents based on dimensions. 

We'll need to tune this based on benchmarks. If we support count and sum as defaults , we can inherently support average. Maybe we can switch to count and sum later if we find that the overhead is high.